pwndbg/hexdump.py

```diff
@@ -66,51 +79,110 @@ def load_color_scheme():
-        if skip and line == last_line:
-            if skipping:
-                continue
+    if not dX_call:
+        if not color_scheme or not printable:
+            load_color_scheme()
+
+        # If there's nothing to print, just print the offset and address and return
+        if len(data) == 0:
+            yield H.offset("+%04x " % len(data)) + H.address("%#08x  " % (address + len(data)))
+
+            # Don't allow iterating over this generator again
+            return
+
+        data = list(bytearray(data))
+        last_line = None
+        skipping = False
+        for i, line in enumerate(groupby(width, data, fill=-1)):
+            if skip and line == last_line:
+                if skipping:
+                    continue
+
+                skipping = True
+                yield "..."
+            else:
+                skipping = False
+                last_line = line
+
+            hexline = []
+
+            hexline.append(H.offset("+%04x " % ((i + offset) * width)))
+            hexline.append(H.address("%#08x  " % (address + (i * width))))
+
+            for group in groupby(group_width, line):
+                group = reversed(group) if flip_group_endianess else group
+                for idx, char in enumerate(group):
+                    if flip_group_endianess and idx == group_width - 1:
+                        hexline.append(H.highlight_group_lsb(color_scheme[char]))
+                    else:
+                        hexline.append(color_scheme[char])
+                    hexline.append(str(config_byte_separator))
+                hexline.append(" ")
 
-            skipping = True
-            yield "..."
-        else:
-            skipping = False
-            last_line = line
-
-        hexline = []
-
-        hexline.append(H.offset("+%04x " % ((i + offset) * width)))
-        hexline.append(H.address("%#08x  " % (address + (i * width))))
-
-        for group in groupby(group_width, line):
-            group = reversed(group) if flip_group_endianess else group
-            for idx, char in enumerate(group):
-                if flip_group_endianess and idx == group_width - 1:
-                    hexline.append(H.highlight_group_lsb(color_scheme[char]))
-                else:
-                    hexline.append(color_scheme[char])
-                hexline.append(str(config_byte_separator))
-            hexline.append(" ")
-
-        hexline.append(H.separator("%s" % config_separator))
-        for group in groupby(group_width, line):
-            for char in group:
-                hexline.append(printable[char])
             hexline.append(H.separator("%s" % config_separator))
+            for group in groupby(group_width, line):
+                for char in group:
+                    hexline.append(printable[char])
+                hexline.append(H.separator("%s" % config_separator))
+
+            yield "".join(hexline)
+
+    else:
+
+        """
```

Comment:
Nit, generally these types of string comments only go at the top of a function, so in this case we should convert it to a normal comment with `#`.